piqe_ocp_lib/api/resources/ocp_health_checker.py

```diff
@@ -0,0 +1,318 @@
+        is_router_healthy = False
+        unhealthy_router_info = dict()
+        is_router_pod_healthy = False
+        is_replicas_count_matching = False
+        unhealthy_pods = dict()
+        pods_response = self.ocp_pod.list_pods_in_a_namespace(namespace="openshift-ingress")
+        for pod in pods_response.items:
+            for condition in pod["status"]["conditions"]:
+                if condition["type"] == "Ready" and condition["status"] == "False":
+                    unhealthy_pods[pod["metadata"]["name"]] = condition["status"]
+
+        if len(unhealthy_pods.values()) == 0:
+            is_router_pod_healthy = True
+
+        unhealthy_router_info.update({"router_pod": unhealthy_pods.keys()})
+        logger.info("Unhealthy router info : %s", unhealthy_router_info)
+        logger.info("Is router pod/s healthy : %s", is_router_pod_healthy)
+
+        logger.info("Check replicas count of openshift router deployment are matching")
+        deployment_response = self.ocp_deployment.list_all_deployments_in_a_namespace(namespace="openshift-ingress")
+        for deployment in deployment_response.items:
+            if deployment.metadata.name == "router-default":
+                replicas = deployment.status.replicas
+                available_replicas = deployment.status.availableReplicas
+                ready_replicas = deployment.status.readyReplicas
+        if replicas == available_replicas == ready_replicas:
+            is_replicas_count_matching = True
+        unhealthy_router_info.update({"router_replicas": is_replicas_count_matching})
+        logger.info("Is router deployment replicas count matching : %s", is_replicas_count_matching)
+
+        logger.info("Check overall health of router operator")
+        if is_router_pod_healthy and is_replicas_count_matching:
+            is_router_healthy = True
+
+        return is_router_healthy, unhealthy_router_info
+
+    def check_image_registry_health(self):
+        """
+        Check openshift cluster image registry
+        - Check if image registry pod is running fine in openshift-image-registry namespace
+        - Check if deployments of image registry has matching number of replicas
+        :return:
+        """
+        logger.info("Check health of openshift image registry pods")
+        is_image_registry_healthy = False
+        unhealthy_image_registry_info = dict()
+        is_image_registry_pods_healthy = False
+        unhealthy_pods = dict()
+        pods_response = self.ocp_pod.list_pods_in_a_namespace(namespace="openshift-image-registry")
+        for pod in pods_response.items:
+            if "cluster-image-registry-operator" in pod.metadata.name or \
+                    "image-registry" in pod.metadata.name:
+                for condition in pod["status"]["conditions"]:
+                    if condition["type"] == "Ready" and condition["status"] == "False":
+                        unhealthy_pods[pod["metadata"]["name"]] = condition["status"]
```

Comment:
Yes, you are absolutely correct. Certain  loops are identical but for ImageRegistry has more checks. if we try to write common method, Challenge is openshift-router namespace only has one pod vs openshift-image-registry namespace has 6-7 pods running. So in case of ImageRegistry, We are only concern about "cluster-image-registry-operator" and "image-registry" for health check.  Also, When we try to find out replica count, We have to iterate through specific deployments and get their replica count, compare available vs current vs ready replica before we mark them as healthy/unhealthy.  Things will get convoluted ad complicated if we try to combine.